cybergis_compute_client/CyberGISCompute.py

```diff
@@ -78,61 +135,26 @@ def login(self, manualLogin=True, verbose=True):
         Todo:
             Document exceptions/errors raised.
         """
+
+        # login via env variable
         if self.jupyterhubApiToken is not None:
             if self.username is None:
-                res = self.client.request('GET', '/user', {"jupyterhubApiToken": self.jupyterhubApiToken})
-                self.username = res['username']
+                self.set_username()
             if verbose:
                 print('🎯 Logged in as ' + self.username)
             return
-
-        # login via env variable
-        envToken = os.getenv('JUPYTERHUB_API_TOKEN')
```

Comment:
I *really* like that this was split up into more manageable, easy to test chunks. Great work there!